src/S3/Input/PutObjectAclRequest.php

```diff
@@ -196,14 +196,59 @@ public function getVersionId(): ?string
         return $this->VersionId;
     }
 
-    public function requestBody(): array
+    public function requestBody(): string
     {
-        $payload = ['Action' => 'PutObjectAcl', 'Version' => '2006-03-01'];
-        if (null !== $this->AccessControlPolicy) {
-            $payload['AccessControlPolicy'] = $this->AccessControlPolicy;
+        $document = new \DOMDocument('1.0', 'UTF-8');
+        $document->formatOutput = true;
+
+        $document->appendChild($root = $document->createElement('AccessControlPolicy'));
+        $root->setAttribute('xmlns', 'http://s3.amazonaws.com/doc/2006-03-01/');
+
+        if (null === $input = $this->AccessControlPolicy) {
+            return $document->saveXML();
+        }
+
+        if (null !== $input_Grants = $input->getGrants()) {
+            foreach ($input_Grants as $input_Grants_item) {
+                $root->appendChild($root_AccessControlList = $document->createElement('AccessControlList'));
+                $input_Grants_item_AccessControlList = $input_Grants_item;
```

Comment:
mhh.. Something is missing. It should generates:
```xml
<AccessControlPolicy xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
   <AccessControlList>
      <Grant>
         <Grantee>
            <DisplayName>string</DisplayName>
            <EmailAddress>string</EmailAddress>
            <ID>string</ID>
            <xsi:type>string</xsi:type>
            <URI>string</URI>
         </Grantee>
         <Permission>string</Permission>
      </Grant>
   </AccessControlList>
   <Owner>
      <DisplayName>string</DisplayName>
      <ID>string</ID>
   </Owner>
</AccessControlPolicy>
```

 (cf https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObjectAcl.html)

Reply:
fixed

Reply:
Cool I did know what was wrong =)